src/migration/MigrationExecutor.ts

```diff
@@ -424,52 +443,139 @@ export class MigrationExecutor {
         const allMigrations = this.getMigrations()
 
         // find the instance of the migration we need to remove
-        const migrationToRevert = allMigrations.find(
-            (migration) => migration.name === lastTimeExecutedMigration!.name,
-        )
-
-        // if no migrations found in the database then nothing to revert
-        if (!migrationToRevert)
-            throw new TypeORMError(
-                `No migration ${lastTimeExecutedMigration.name} was found in the source code. Make sure you have this migration in your codebase and its included in the connection options.`,
+        const migrationsToRevert = toRevert.map((m) => {
+            const match = allMigrations.find(
+                (migration) => migration.name === m.name,
             )
 
+            if (!match) {
+                // if any of the migrations is found in the database, but not in the source code, throw error to prevent inconsistencies
+                throw new TypeORMError(
+                    `Migration ${
+                        m.name ?? ""
+                    } was not found in the source code. Make sure you have this migration in your codebase and its included in the connection options.`,
+                )
+            }
+
+            return match
+        })
+
         // log information about migration execution
         this.connection.logger.logSchemaBuild(
             `${executedMigrations.length} migrations are already loaded in the database.`,
         )
         this.connection.logger.logSchemaBuild(
             `${
-                lastTimeExecutedMigration.name
+                migrationsToRevert[0].name
             } is the last executed migration. It was executed on ${new Date(
-                lastTimeExecutedMigration.timestamp,
+                migrationsToRevert[0].timestamp,
             ).toString()}.`,
         )
-        this.connection.logger.logSchemaBuild(`Now reverting it...`)
+        this.connection.logger.logSchemaBuild(
+            `Reverting following migrations: ${migrationsToRevert
+                .map((m) => m.name)
+                .join(", ")}`,
+        )
+
+        if (this.transaction === "all") {
+            // If we desire to revert all migrations in a single transaction
+            // but there is a migration that explicitly overrides the transaction mode
+            // then we have to fail since we cannot properly resolve that intent
+            // In theory we could support overrides that are set to `true`,
+            // however to keep the interface more rigid, we fail those too
+            const migrationsOverridingTransactionMode =
+                migrationsToRevert.filter(
+                    (migration) =>
+                        !(migration.instance?.transaction === undefined),
```

Comment:
```suggestion
                        migration.instance?.transaction !== undefined,
```

Reply:
OR

```suggestion
                        migration.instance?.transaction != null,
```

Reply:
Like most of the projects nowadays, we use strict equality checks (`===` and `!==` instead of `==` and `!=`) 😉

Not sure if it's enforced with the eslint rule `eqeqeq`, I think we just do it.

Reply:
FWIW I am personally a fan of using loose equality (`== null` or `!= null`) to check for the absence of a value in cases where you don't care to distinguish between `null` and `undefined` and where both values could occur.

Unlike loose equality comparisons involving other types, which can be treacherous, `null` is only loosely equal to itself and `undefined` so this is quite safe.

By contrast, rigidly using only strict equality requires that you remember to check for both values (`x === null || x === undefined`) and so is more likely to lead to errors as well as being less concise.

The `eqeqeq` eslint rule has [an option](https://eslint.org/docs/latest/rules/eqeqeq#allow-null) to support this convention. It looks like we aren't enabling `eqeqeq` at all in TypeORM currently though.